TestApplicationCreateCommand "tail logs" test is flaky

This failure occurs periodically in CI

```
    --- FAIL: TestApplicationCreateCommand/tail_timeout (0.03s)
##[error]        command_table.go:366: Unexpected output (-expected, +actual):   strings.Join({
              	`Created application "my-application"`,
            - 	"...log output...",
              	`Timeout after "5ms" waiting for "my-application" to become ready`,
              	"To view status run: riff application list --namespace default",
              	... // 2 identical lines
              }, "\n")
##[error]        application_create_test.go:964: FAIL:	ApplicationLogs(string,*v1alpha1.Application,time.Duration,string)
            		at: [application_create_test.go:936 command_table.go:218]
##[error]        application_create_test.go:964: FAIL: 0 out of 1 expectation(s) were met.
            	The code you are testing needs to make 1 more call(s).
            	at: [application_create_test.go:964 command_table.go:211 command_table.go:373]
```

https://github.com/projectriff/cli/runs/331511597

--- a/pkg/race/runner.go
+++ b/pkg/race/runner.go
@@ -19,6 +19,8 @@ package race
 import (
 	"context"
 	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 // Task is a function which performs some kind of activity and returns an error value. It also observes the given
@@ -26,24 +28,25 @@ import (
 type Task func(ctx context.Context) error
 
 // Run runs each task in its own go routine and returns the result of the first task to complete (or error). A timeout
-// is specified to limit the whole process. Each task should observe the context and return once the context is done.
+// is specified to limit the whole process. Each task must observe the context and return once the context is done.
 func Run(ctx context.Context, timeout time.Duration, tasks ...Task) error {
 	ctx, cancel := context.WithTimeout(ctx, timeout)
-	defer cancel()
 	errChan := make(chan error, len(tasks)+1)
 	defer close(errChan)
 
+	var wg wait.Group
 	for _, task := range tasks {
-		go func(task Task) {
+		task := task
+		wg.StartWithContext(ctx, func(ctx context.Context) {
 			defer cancel()
 			err := task(ctx)
 			if ctx.Err() == nil {
 				errChan <- err
 			}
-		}(task)
+		})
 	}
 
-	<-ctx.Done()
+	wg.Wait()
 	errChan <- ctx.Err()
 
 	return <-errChan

--- a/pkg/race/runner_test.go
+++ b/pkg/race/runner_test.go
@@ -34,7 +34,6 @@ func TestRun(t *testing.T) {
 	}{{
 		name:    "empty",
 		timeout: time.Millisecond,
-		err:     context.DeadlineExceeded,
 	}, {
 		name:    "return immediately",
 		timeout: time.Minute,

